### PR TITLE
web: No context menu on iOS if finger moves too far on long-press

### PR DESCRIPTION
**Pull request by n0samu**
Supersedes #10691.

### CODE REVIEW

File: web/packages/core/src/ruffle-player.ts [hunk 1]
```diff
@@ -816,6 +822,21 @@ export class RufflePlayer extends HTMLElement {
...
+    /**
+     * Helper function to get the distance between two points
+     *
+     * @param x1 The x coordinate of point 1.
+     * @param y1 The y coordinate of point 1.
+     * @param x2 The x coordinate of point 2.
+     * @param y2 The y coordinate of point 2.
+     * @returns The distance between the two points.
+     */
+    getDistance(x1: number, y1: number, x2: number, y2: number): number {
```
Comment (relrelb) [inline web/packages/core/src/ruffle-player.ts:834]: The logic looks good in general, thank you!

One nitpick - I think the more TypeScripty way to handle points would be to introduce:
```ts
class Point {
    constructor(public x: number, public y: number) {}

    distanceTo(other: Point) {
        const dx = other.x - this.x;
        const dy = other.y - this.y;
        return Math.sqrt(dx * dx + dy * dy);
    }
}
```
Then use it in the various places. What do you think?

Comment (n0samu) [reply web/packages/core/src/ruffle-player.ts]: I agree that that's cleaner. Added, thanks!

File: web/packages/core/src/ruffle-player.ts [hunk 2]
```diff
@@ -148,6 +158,8 @@ export class RufflePlayer extends HTMLElement {
 
     private isExtension = false;
     private longPressTimer: ReturnType<typeof setTimeout> | null = null;
+    private pointerDownPosition = new Point(-1, -1);
```
Comment (relrelb) [inline web/packages/core/src/ruffle-player.ts:161]: I don't know if this makes sense, but initalizing this with `null` seems more clear to me. Then a non-null assertion will be needed in `checkLongPressMovement`.

Comment (n0samu) [reply web/packages/core/src/ruffle-player.ts]: Ok, updated

**Review by relrelb — APPROVED** [review]

File: web/packages/core/src/ruffle-player.ts (hunk 2 above)
Comment (n0samu) [reply web/packages/core/src/ruffle-player.ts]: ... and resolved linter's complaint, oops

**Pull request merged**